cluster-operator/src/main/java/io/strimzi/operator/cluster/model/cruisecontrol/Capacity.java

```diff
@@ -219,26 +219,29 @@ private static CpuCapacity getCpuBasedOnRequirements(ResourceRequirements resour
         Integer limit = getResourceRequirement(resourceRequirements, ResourceRequirementType.LIMIT);
 
         if (limit != null) {
-            if (request == null || limit.intValue() == request.intValue()) {
+            if (request != null) {
+                return new CpuCapacity(CpuCapacity.milliCpuToCpu(request));
+            } else {
                 return new CpuCapacity(CpuCapacity.milliCpuToCpu(limit));
             }
+        } else if (request != null) {
+            return new CpuCapacity(CpuCapacity.milliCpuToCpu(request));
         }
-
         return null;
     }
 
-    private CpuCapacity processCpu(io.strimzi.api.kafka.model.balancing.BrokerCapacity bc, BrokerCapacityOverride override, CpuCapacity cpuBasedOnRequirements) {
-        if (cpuBasedOnRequirements != null) {
-            if ((override != null && override.getCpu() != null) || (bc != null && bc.getCpu() != null)) {
-                LOGGER.warnCr(reconciliation, "Ignoring CPU capacity override settings since they are automatically set to resource limits");
-            }
-            return cpuBasedOnRequirements;
-        } else if (override != null && override.getCpu() != null) {
+    private CpuCapacity processCpu(BrokerCapacityOverride override, io.strimzi.api.kafka.model.balancing.BrokerCapacity bc, ResourceRequirements resourceRequirements) {
+        if (override != null && override.getCpu() != null) {
             return new CpuCapacity(override.getCpu());
         } else if (bc != null && bc.getCpu() != null) {
             return new CpuCapacity(bc.getCpu());
         } else {
-            return new CpuCapacity(BrokerCapacity.DEFAULT_CPU_CORE_CAPACITY);
+            CpuCapacity cpu = getCpuBasedOnRequirements(resourceRequirements);
+            if (cpu != null) {
+                return cpu;
+            } else {
+                return new CpuCapacity(BrokerCapacity.DEFAULT_CPU_CORE_CAPACITY);
```

Comment:
Is there a reason why `getCpuBasedOnRequirements` returns null and not `new CpuCapacity(BrokerCapacity.DEFAULT_CPU_CORE_CAPACITY)`?

Reply:
Refactored to return `new CpuCapacity(BrokerCapacity.DEFAULT_CPU_CORE_CAPACITY)`